packages/release-mastermind/src/contexts/issues.ts

```diff
@@ -8,58 +1,154 @@
+			throw err
+		})
+
+		const currentVersion: Version = await utils.versioning
+			.parse(config, config.issue?.ref)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while parsing versioning: `, err)
+				throw err
+			})
+
+		return {
+			sha: context.sha,
+			action: context.payload.action as string,
+			currentVersion,
+			IDNumber: context.payload.issue?.id,
+			props: {
+				type: "issue",
+				ID: issue.number,
+				creator: issue.user.login,
+				description: issue.body || "",
+				locked: issue.locked,
+				state: issue.state,
+				labels,
+				title: issue.title
+			}
+		}
+	}
+
+	async run(attempt?: number) {
+		if (!this.config)
+			throw new LoggingDataClass(
+				LoggingLevels.warn,
+				"Cannot start without config"
+			)
+		if (!attempt) {
+			attempt = 1
+			core.startGroup("Issue Actions")
+		}
+		let seconds = attempt * 10,
+			enforceConventionsSuccess = true
+		try {
+			if (this.config.enforceConventions && this.util.shouldRun("convention"))
+				enforceConventionsSuccess = await this.conventions.enforce(this)
+
+			if (enforceConventionsSuccess) {
+				if (this.config.labels && this.util.shouldRun("label"))
+					await this.applyLabels(this).catch((err) => {
+						log(LoggingLevels.error, "Error applying label", err)
+					})
+
+				if (this.config.assignProject && this.util.shouldRun("release"))
+					await this.assignProject(this).catch((err) => {
+						log(LoggingLevels.error, "Error assigning projects", err)
+					})
 
-  async run(attempt?: number) {
-    if (!this.config)
-      throw new LoggingDataClass(LoggingLevels.warn, 'Cannot start without config')
-    if (!attempt) {
-      attempt = 1
-      core.startGroup('Issue Actions')
-    }
-    let seconds = attempt * 10,
-      enforceConventionsSuccess: boolean = true
-    try {
-      if (this.config.enforceConventions)
-        enforceConventionsSuccess = await this.conventions.enforce(this)
-      if (enforceConventionsSuccess) {
-        if (this.config.labels)
-          await this.applyLabels(this).catch(err => {
-            log(LoggingLevels.error, 'Error applying label', err)
-          })
-        if (this.config.assignProject)
-          await this.assignProject(this).catch(err => {
-            log(LoggingLevels.error, 'Error assigning projects', err)
-          })
-        core.endGroup()
-      }
-    } catch (err) {
-      if (attempt > 3) {
-        core.endGroup()
-        throw new LoggingDataClass(LoggingLevels.emergency, `Issue actions failed. Terminating job.`)
-      }
-      log(
-          LoggingLevels.warn,
-          `Issue Actions failed with "${err}", retrying in ${seconds} seconds....`
-      )
-      attempt++
-      setTimeout(async () => {
-        this.run(attempt)
-      }, seconds * 1000)
-    }
-  }
+				core.endGroup()
+			}
+		} catch (err) {
+			if (attempt > this.retryLimit) {
+				core.endGroup()
+				throw new LoggingDataClass(
+					LoggingLevels.emergency,
+					`Issue actions failed. Terminating job.`
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Videndum/action-masterminds/pullRequest?prid=7872005)
```suggestion
          "Issue actions failed. Terminating job."
```